### PR TITLE
V13: Webhook documentation

### PR DESCRIPTION
**Pull request by Zeegaan**
## Description

- Added documentation for the new Webhooks feature in v13 🚀 

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
V13+

## Deadline (if relevant)
Tomorrow 🙈

### CODE REVIEW

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+﻿---
+description: Get started with Webhooks
+---
+
+# Webhooks
+
+Webhooks provide real-time, event-driven communication within Umbraco. Seamlessly integrated, these lightweight, HTTP-based notifications empower you to trigger instant actions and synchronize data.  And with its different extension points, you can tailor these webhooks. to fit a broad range of requirements.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:7]:
```suggestion
Webhooks provide real-time, event-driven communication within Umbraco. Seamlessly integrated, these lightweight, HTTP-based notifications empower you to trigger instant actions and synchronize data. With its different extension points, you can tailor these webhooks to fit a broad range of requirements.
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+## Getting Started
+
+To work with Webhooks, you need to go to the Webhooks within the Settings section:
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:11]:
```suggestion
To work with Webhooks, you need to go to Webhooks within the Settings section:
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+![Webhooks section](images/webhook-section.png)
+
+From here we can create a webhook, by clicking the `Create webhook` button, this will take you to the create webhook screen:
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:14]:
```suggestion
From here we can create a webhook, by clicking the `Create webhook` button, which will take you to the Create webhook screen:
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+![Creating a webhook](images/create-webhook.png)
+
+## Url
+The `Url` should be the endpoint you want the webhook to send a request to, whenever a given `Event` is fired.
+
+## Events
+Events are when a given action happens, by default there are 5 events you can choose from.
+
+- Content Published - This event happens whenever a content gets published.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:24]:
```suggestion
- Content Published - This event happens whenever some content gets published.
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+- Content Unpublished - This event happens whenever a content gets unpublished
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:25]:
```suggestion
- Content Unpublished - This event happens whenever some content gets unpublished
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+- Content Deleted - This event happens whenever a content gets deleted.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:26]:
```suggestion
- Content Deleted - This event happens whenever some content gets deleted.
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+- Media Deleted - This event happens whenever a media item is deleted.
+- Media Saved - This event happens whenever a media item is saved.
+
+## Content type
+If you have selected a Content / Media event, you can specify if you only want your webhook to fire with a given Document/Media type.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:31]:
```suggestion
If you have selected a Content or Media event, you can specify if you only want your webhook to fire with a given Document or Media type.
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+For example, if you have selected `Content Published` event. You can then specify that you only want the webhook to fire, when the content is of a given content type.
+
+## Headers
+You can specify custom headers, that will be sent with your request.
+
+For example you could specify `Accept: application/json`, security headers, etc.
+
+# Defaults
+Umbraco webhooks have been configured with some defaults, such as default headers or some events send a payload. In this section we will take a look at those.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:41]:
```suggestion
Umbraco webhooks have been configured with some defaults, such as default headers or some events that send a payload. In this section, we will take a look at those.
```

File: 13/umbraco-cms/reference/webhooks/README.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+      "Path": "root"
+    }
+  },
+  "Id": "c1922956-7855-4fa0-8f2c-7af149a92135",
+  "ContentType": "root",
+  "Properties": {}
+}
+```
+
+But the `Content deleted` does not send the entire content as json, instead it sends the `Id` of the content like so:
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/README.md:63]:
```suggestion
however, the `Content deleted` does not send the entire content as JSON, instead, it sends the `Id` of the content like so:
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+With Umbraco, you can even create your own webhook events.
+
+This documentation guides you through the process of implementing your own webhook events using the `WebhookEventBase<TNotification>` base class.
+
+## Creating an event with the WebhookEventBase
+
+The `WebhookEventBase<TNotification>` class serves as the foundation for creating custom webhook events. Here's a brief overview of its key components:
+
+- **Alias**: Property that must be overridden to provide a unique identifier for your webhook event.
+- **EventName**: Property that represents the name of the event. It is automatically set based on the provided alias unless explicitly specified.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:18]:
```suggestion
- **EventName**: A property that represents the name of the event. It is automatically set based on the provided alias unless explicitly specified.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+- **EventType**: Property that categorizes the event type. It defaults to "Others" but can be customized using the `WebhookEventAttribute`.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:19]:
```suggestion
- **EventType**: A property that categorizes the event type. It defaults to "Others" but can be customized using the `WebhookEventAttribute`.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+- **WebhookSettings**: Property containing the current webhook settings.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:20]:
```suggestion
- **WebhookSettings**: The property containing the current webhook settings.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+- **ProcessWebhooks**: Method responsible for processing webhooks for a given notification.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:21]:
```suggestion
- **ProcessWebhooks**: The method responsible for processing webhooks for a given notification.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+- **ShouldFireWebhookForNotification**: Method determining whether webhooks should be fired for a specific notification.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:22]:
```suggestion
- **ShouldFireWebhookForNotification**: The method determining whether webhooks should be fired for a specific notification.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+- **ConvertNotificationToRequestPayload**: Optional method allowing customization of the notification payload before sending it to webhooks.
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:23]:
```suggestion
- **ConvertNotificationToRequestPayload**: An optional method allowing customization of the notification payload before sending it to webhooks.
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+1. **Derive from `WebhookEventBase<TNotification>`**:
+
+    ```csharp
+    public class YourCustomEvent : WebhookEventBase<YourNotificationType>
+    {
+        // Constructor and required overrides go here
+    }
+    ```
+
+2. **Override Alias Property**:
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:38]:
```suggestion
2. **Override the Alias Property**:
```

File: 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+
+1. **Derive from `WebhookEventContentBase<TNotification, TEntity>`**:
+
+    ```csharp
+    public class YourContentWebhookEvent : WebhookEventContentBase<YourNotificationType, YourContentBaseType>
+    {
+    }
+    ```
+
+2. **Override Required Methods**:
```
Comment (jonat123) [inline 13/umbraco-cms/reference/webhooks/expanding-webhook-events.md:135]:
```suggestion
2. **Override the Required Methods**:
```

**Comment by Zeegaan** [comment]
@jonat123 done 🚀

**Review by jonat123 — APPROVED** [review]

**Pull request merged**